npm/pkg/dataplane/types.go

```diff
@@ -76,3 +78,70 @@ func (npmPod *updateNPMPod) updateIPSetsToRemove(setNames []*ipsets.IPSetMetadat
 		npmPod.IPSetsToRemove = append(npmPod.IPSetsToRemove, set.GetPrefixName())
 	}
 }
+
+type updatePodCache struct {
+	sync.Mutex
+	cache map[string]*updateNPMPod
+	// queue maintains the FIFO queue of the pods in the cache.
+	// This makes sure we handle issue 1729 with the same queue as the control plane.
+	// It also lets us update Pod ACLs in the same queue as the control plane so that
+	// e.g. the first Pod created is the first Pod to have proper connectivity.
+	queue           []string
+	initialCapacity int
+}
+
+// newUpdatePodCache creates a new updatePodCache with the given initial capacity for the queue
+func newUpdatePodCache(initialCapacity int) *updatePodCache {
+	return &updatePodCache{
+		cache:           make(map[string]*updateNPMPod),
+		queue:           make([]string, 0, initialCapacity),
+		initialCapacity: initialCapacity,
+	}
+}
+
+// enqueue adds a pod to the queue if necessary and returns the pod object used
+func (c *updatePodCache) enqueue(m *PodMetadata) *updateNPMPod {
+	pod, ok := c.cache[m.PodKey]
+	if !ok {
```

Comment:
What if the nodename got updated ? we will need to check if the pod objs are same if not update the element in map and move the pod obj to the end ?